src/charm.py

```diff
@@ -841,6 +811,20 @@ def _postgresql_layer(self) -> Layer:
                 },
             },
         }
+        if "tls" not in self.unit_peer_data:
+            layer_config.update(
+                {
+                    "checks": {
+                        self._postgresql_service: {
+                            "override": "replace",
+                            "level": "ready",
+                            "http": {
+                                "url": f"{self._patroni._patroni_url}/health",
```

Comment:
Hm, from the README, `http` check should work with https too.
https://github.com/canonical/pebble/blob/master/README.md?plain=1#L557-L559

If it doesn't, maybe we should check with the `exec curl {self._patroni._patroni_url}/health` command always.

Reply:
Updated on https://github.com/canonical/postgresql-k8s-operator/pull/134/commits/364adae7f174bbfe307007e3facd5458e518d5c7. Now TLS is working. Thanks to @dragomirp.